ksql-engine/src/main/java/io/confluent/ksql/metrics/ProducerCollector.java

```diff
@@ -107,12 +107,30 @@ public void close() {
     topicSensors.values().forEach(v -> v.close(metrics));
   }
 
+  @Override
   public Collection<TopicSensors.Stat> stats(String topic) {
     final List<TopicSensors.Stat> list = new ArrayList<>();
     topicSensors.values().stream().filter(counter -> counter.isTopic(topic)).forEach(record -> list.addAll(record.stats()));
     return list;
   }
 
+  @Override
+  public double currentMessageConsumptionRate() {
```

Comment:
You can put default impls into the MetricCollector interface.

default double currentMessageConsumptionRate() {
        return 0;
}